core/src/test/java/apoc/periodic/PeriodicTest.java

```diff
@@ -216,6 +219,42 @@ public void testTerminateIterate() throws Exception {
         PeriodicTestUtils.testTerminatePeriodicQuery(db, "CALL apoc.periodic.iterate('UNWIND range(0,1000) as id RETURN id', 'WITH $id as id CREATE (:Foo {id: $id})', {batchSize:10,iterateList:false})");
     }
 
+    @Test
+    public void testTerminateIterateWithTerminateTransactionCommand()  {
+        // apoc.periodic.iterate
+        PeriodicTestUtils.testTerminateWithCommand(db, "CALL apoc.periodic.iterate('UNWIND range(0,9999999) as id RETURN id', 'WITH $id as id CREATE (:Foo {id: $id})', {batchSize:1,parallel:true})");
+        PeriodicTestUtils.testTerminateWithCommand(db, "CALL apoc.periodic.iterate('UNWIND range(0,9999999) as id RETURN id', 'WITH $id as id CREATE (:Foo {id: $id})', {batchSize:10,iterateList:true})");
+        PeriodicTestUtils.testTerminateWithCommand(db, "CALL apoc.periodic.iterate('UNWIND range(0,9999999) as id RETURN id', 'WITH $id as id CREATE (:Foo {id: $id})', {batchSize:10,iterateList:false})");
+        PeriodicTestUtils.testTerminateWithCommand(db, "CALL apoc.periodic.iterate('CALL apoc.util.sleep(19999) RETURN 1 as id', 'WITH $id as id CREATE (:Foo {id: $id})', {batchSize:10,iterateList:false})");
+        
+        // apoc.periodic.commit
+        PeriodicTestUtils.testTerminateWithCommand(db, "CALL apoc.periodic.commit('UNWIND range(0,999999) as id WITH id CREATE (n:Foo {id: id}) RETURN n limit 1000', {})");
+    }
+    
+    @Test
+    public void testWithTerminationInnerTransaction() {
+        // terminating the apoc.util.sleep should instantly terminate the periodic query without any creation
+        final String innerLongQuery = "CALL apoc.util.sleep(99999) RETURN 0";
+        final String query = "CALL apoc.periodic.iterate($innerQuery, 'WITH $id as id CREATE (:Foo {id: $id})', {params: {innerQuery: $innerQuery}})";
+
+        terminateTransactionAsync(db, innerLongQuery);
+
+        // assert query terminated (RETURN 0 nodesCreated)
+        try {
+            TestUtil.testCall(db, query,
+                    Map.of("innerQuery", innerLongQuery),
+                    row -> {
+                        final Object actual = ((Map) row.get("updateStatistics")).get("nodesCreated");
+                        assertEquals(0L, actual);
+                    });
+            fail("Should have terminated");
```

Comment:
What does this fail give us?

Reply:
It is just to verify that the query really fails, because of terminationGuard, and don't go ahead without errors.